src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/entity/VariationLicence.kt

```diff
@@ -303,37 +308,40 @@ class VariationLicence(
   ) = copy(
     statusCode = statusCode,
     dateLastUpdated = LocalDateTime.now(),
-    updatedByUsername = updatedByUsername,
+    updatedByUsername = staffMember?.username ?: SYSTEM_USER,
     approvedByUsername = approvedByUsername,
     approvedByName = approvedByName,
     approvedDate = approvedDate,
     supersededDate = supersededDate,
     submittedDate = submittedDate,
     licenceActivatedDate = licenceActivatedDate,
+    updatedBy = staffMember ?: this.updatedBy,
   )
 
   override fun overrideStatus(
     statusCode: LicenceStatus,
-    updatedByUsername: String?,
+    staffMember: Staff?,
```

Comment:
I don't think this will ever be a staffMember, as it's our support team that do these overrides. Unless they are also present in the staff table, in which case the opposite is true and it should always be populated and shouldn't default IMO

Reply:
Confirmed, logic should apply in all places we are setting username

Reply:
I think all our staff are nomis users so will be recorded in the staff table.